File: src/test/java/com/qomolangma/api/PingTest.java
package com.qomolangma.api;

import com.qomolangma.frameworks.test.web.IntegrationTest;
import com.qomolangma.frameworks.test.web.PathVariables;
import io.restassured.response.ValidatableResponse;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static org.hamcrest.Matchers.is;

public class PingTest extends IntegrationTest {
    @Test
    void should_reply_pong() {
        ValidatableResponse post = post("/ping", PathVariables.variables(Map.of()), Map.of("data", "pong"), null);
        post.statusCode(is(200)).body("ping", is("pong")).body("message", is("ok"));
    }
}

File: src/test/kotlin/com/qomolangma/api/PingTest.kt
package com.qomolangma.api

import com.qomolangma.frameworks.test.web.Documentation
import com.qomolangma.frameworks.test.web.IntegrationTest
import com.qomolangma.frameworks.test.web.PathVariables.Companion.variables
import org.hamcrest.Matchers.`is`
import org.junit.jupiter.api.Test
import org.springframework.restdocs.payload.PayloadDocumentation.*

class PingTest : IntegrationTest() {
    @Test
    fun should_reply_pong() {
        val post = post("/ping", variables(HashMap<String, Any>()), mapOf(Pair("data", "pong")), documentation())
        post.statusCode(`is`(200)).body("ping", `is`("pong")).body("message", `is`("ok"))
    }

    fun documentation(): Documentation {
        return Documentation.doc(
            "test.ping", requestFields(
                fieldWithPath("data").description("测试数据")
            ), responseFields(
                fieldWithPath("ping").description("ping"),
                fieldWithPath("message").description("message")
            )
        )
    }
}
